python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -496,7 +508,11 @@ def any_deps_updated() -> "DepsAutomationCondition":
         """
         with disable_dagster_warnings():
             return AutomationCondition.any_deps_match(
-                AutomationCondition.newly_updated() | AutomationCondition.will_be_requested()
+                (
+                    AutomationCondition.newly_updated()
+                    & ~AutomationCondition.executed_with_root_target()
```

Comment:
Can you make the comment about perf you made in the PR discussion as a comment here.